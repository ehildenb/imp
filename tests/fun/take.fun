// [ 1 : 2 : 3 : 4 : 5 : .Vals ]

letrec
    cons h [ t ] = [ h : t ]
and take n = fun [       ] -> [ ]
             |   [ h : t ] -> if n > 0 then cons h (take (n - 1) [ t ]) else [ ]
 in take 5 [ 1 : 2 : 3 : 4 : 5 : 6 : 7 : .Vals ]
