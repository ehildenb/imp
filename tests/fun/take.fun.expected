<FUN>
  <k>
    [ 1 : 2 : 3 : 4 : 5 : .Vals ]
  </k>
  <env>
    .Map
  </env>
  <envs>
    .List
  </envs>
</FUN>
